### PR TITLE
Pathfinding

### PR DESCRIPTION
**Pull request by OliverW10**
Adds pathfinding utility and swaps movement trajectory generation to useing pathplanner rather than the wpilib trajectories

### CODE REVIEW

File: utilities/pathfinding.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        self.y_max += amount
+
+    def get_corners(self) -> list[Pose2d]:
+        c1 = Pose2d(self.x_min, self.y_min, 0)
+        c2 = Pose2d(self.x_max, self.y_min, 0)
+        c3 = Pose2d(self.x_max, self.y_max, 0)
+        c4 = Pose2d(self.x_min, self.y_max, 0)
+        return [c1, c2, c3, c4]
+
+    def __eq__(self, __o: object) -> bool:
```
Comment (auscompgeek) [inline utilities/pathfinding.py:67]: Why are we defining our own `__eq__` instead of the generated one?

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 numpy~=1.24.1
-robotpy[apriltag,cscore]~=2023.3.2
+robotpy[apriltag,cscore,pathplannerlib]~=2023.3.2
```
Comment (auscompgeek) [inline requirements.txt:2]: I thought we wanted to only use the `robotpy` package to pin WPILib packages?

File: requirements.txt [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
 robotpy-ctre~=2023.0.0
 robotpy-navx~=2023.0.3
 robotpy-rev~=2023.1.3.1
 robotpy-photonvision~=2023.3.0
+astar
```
Comment (auscompgeek) [inline requirements.txt:7]: Please keep this in alphabetical order.

File: requirements.txt [hunk 4]
```diff
@@ -1,6 +1,7 @@
 numpy~=1.24.1
```
Comment (auscompgeek) [inline requirements.txt:1]: Pipfile needs to be updated at the same time.

**Review by auscompgeek — CHANGES_REQUESTED** [review]
See inline comments

**Comment by LucienMorey** [comment]
Probs not going to use it ay

**Pull request closed without merging**